### PR TITLE
Login view.

### PR DESCRIPTION
**Pull request by abvg9**
Adding login view.

 - [x] String validation logic.
 - [x] Create templates for visual elements of the application.
 - [x] Implement login view base.
 - [x] Add Google login.
 - [x] Add Facebook login.
 - [x] Implement inheritance for the different ways of login.
 - [x] Implement http pettitions provider.
 - [x] Implement final desing.
 - [x] Implement list view of children.
 - [x] Implement list view of groups.
 - [x] Implement Splash screen.
 - [x] Implement tests.

### CODE REVIEW

File: lib/view/widget/button/pictron_base_button.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    Key key,
+    String text = "",
+    Function function,
+    TextStyle textStyle,
+  }) : super(
+          children: <Widget>[
+            RaisedButton(
+                onPressed: () {
+                  function();
+                },
```
Comment (notaphplover) [inline lib/view/widget/button/pictron_base_button.dart]: Maybe there is a better way to express this:

```dart
    children: <Widget>[
            RaisedButton(
                onPressed: function,
...
```

Lets try to create self descriptive variables. The name "function" is not as descriptive as "onPressed".

File: lib/view/widget/button/pictron_base_button.dart [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+                child: Text(text, style: textStyle))
```
Comment (notaphplover) [inline lib/view/widget/button/pictron_base_button.dart]: It's a bit tricky to have two closing parentheses characters here. The last one closes the RaisedButton element, so it should be at a new line.

File: lib/view/widget/text/pictron_password_text_field.dart [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+class PictronPasswordTextField extends PictronBaseTextField {
+  PictronPasswordTextField({
+    Key key,
+    String placeholder,
+  }) : super(
+    key: key,
+    placeholder: placeholder,
+    obscureText: true,
+  );
+}
```
Comment (notaphplover) [inline lib/view/widget/text/pictron_password_text_field.dart]: Let's try to add Line feeds at the end of the files. See the following [page](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline).

**Review by FRYoussef — APPROVED** [review]
Go ahead.

**Pull request merged**